api.py

```diff
@@ -71,14 +122,131 @@ def get(self):
 		self.response.headers['Content-Type'] = 'application/json'
 		self.response.write(json.dumps(output))
 
+class RankAPI(webapp2.RequestHandler):
+	def get(self, rank_type, id):
+		if not check_authentication(self):
+			return
+
+		member = Member.query(Member.id == id).get()
+		if not member:
+			# 404 if a nonexistent member is specified
+			self.error(404)
+			return
+
+		selected_semester = self.request.get('semester')
+		
+		if not selected_semester:
+			selected_semester = get_current_semester()
+		else:
+			# Avoid throwing error 500 if a bad semester string is supplied
+			try:
+				selected_semester = float(selected_semester)
+			except ValueError:
+				self.error(400)
+				return
+		
+		if rank_type == 'name':
```

Comment:
Instead of returning rank types, what about returning a Rank object, like how ranks are defined in `ranks.py`, that just includes all three?  You already set the output type as JSON, so it does not need to just be one string.

Reply:
Good point. Changed to return an object with all 3 options rather than selecting one